h5grove/encoders.py

```diff
@@ -104,26 +103,39 @@ def encode(content, encoding: Optional[str] = "json") -> Response:
         - `csv`: nD arrays in downloadable csv files
         - `npy`: nD arrays in downloadable npy files
         - `tiff`: 2D arrays in downloadable TIFF files
+    :param dtype: Data type conversion
+        - `origin` (default): No conversion
+        - `safe`: Convert to a type supported by JS typedarray
+        Only supported for nD array/scalars.
     :returns: A Response object containing content and headers
     :raises ValueError: If encoding is not among the ones above.
     """
+    if dtype in ("origin", None):
+        cast_content = content
+    elif dtype == "safe":
+        cast_content = sanitize_array(content, copy=False)
+    else:
+        raise ValueError(f"Unsupported dtype {dtype}")
+
     if encoding in ("json", None):
         return Response(
-            orjson_encode(content),
+            orjson_encode(cast_content),
             headers={"Content-Type": "application/json"},
         )
 
+    content_array = np.array(cast_content, copy=False)
```

Comment:
Convert to `ndarray` here since following `*_encode` functions expect a `ndarray` as argument.

Reply:
Same here. Rather than converting blindly and hope for the best, we should check the type of `cast_content`.